### PR TITLE
Add support for property annotations with the experimental decorator

### PR DESCRIPTION
**Pull request by BenWilson2**
Signed-off-by: Ben Wilson <benjamin.wilson@databricks.com>

## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Modify the patch behavior of the experimental annotation decorator to support the sphinx expected formatting of property type pydoc strings.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests (describe details, including test results, below)

Local build of the docs to verify that the pydoc modification works for property type methods

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by BenWilson2** [comment]
Screen Shot 2023-01-04 at 10 48 55 AM
Annotation for property methods renders correctly as an append to the pydoc string (cannot prepend in sphinx to a property type pydoc)

File: mlflow/models/model.py [hunk 1]
```diff
@@ -68,37 +70,55 @@ def __init__(
...
 
     @property
     def model_uri(self):
         """
         The ``model_uri`` of the logged model in the format ``'runs:/<run_id>/<artifact_path>'``.
+
+        :getter: Gets the uri path of the logged model from the uri `runs:/<run_id>` path
+            encapsulation
+        :setter: Sets the run_id-centric uri path for the model
+        :type: str
```
Comment (harupy) [inline mlflow/models/model.py:121]: Do we have a setter for `model_uri`?

Comment (harupy) [reply mlflow/models/model.py:121]: Does `ModelInfo_instance.model_uri = "my-uri"` work?

File: mlflow/models/model.py [hunk 2]
```diff
@@ -107,6 +127,10 @@ def model_uuid(self):
         """
         The ``model_uuid`` of the logged model,
         e.g., ``'39ca11813cfc46b09ab83972740b80ca'``.
+
+        :getter: [Legacy] Gets the model_uuid (run_id) of a logged model
+        :setter: [Legacy] Sets the model_uuid (run_id) of a model
```
Comment (WeichenXu123) [inline mlflow/models/model.py:132]: Similar question with above, _model_uuid should be immutable once model was logged ?

**Comment by BenWilson2** [comment]
Yeah, not sure why I applied those setters to all of those properties. Removed them and adjusted the property definitions to what they should be.

File: mlflow/models/model.py [hunk 3]
```diff
@@ -68,37 +70,51 @@ def __init__(
     def artifact_path(self):
         """
         Run relative path identifying the logged model.
+
+        :getter: Retrieves the relative path of the logged model.
+        :type: str
+
```
Comment (harupy) [inline mlflow/models/model.py:76]:
```suggestion
```

File: mlflow/models/model.py [hunk 4]
```diff
@@ -36,6 +37,7 @@
 class ModelInfo:
     """
     The metadata of a logged MLflow Model.
+
```
Comment (harupy) [inline mlflow/models/model.py:40]:
```suggestion
```

File: mlflow/models/model.py [hunk 5]
```diff
@@ -131,6 +156,9 @@ def signature_dict(self):
         """
         A dictionary that describes the model input and output generated by
         :py:meth:`ModelSignature.to_dict() <mlflow.models.ModelSignature.to_dict>`.
+
+        :getter: Gets the model signature as a dictionary
+        :type: Dict(str, Any)
```
Comment (harupy) [inline mlflow/models/model.py:161]:
```suggestion
        :type: Dict[str, Any]
```

File: mlflow/models/model.py [hunk 6]
```diff
@@ -298,16 +346,29 @@ def metadata(self) -> Optional[Dict[str, Any]]:
...
     def signature(self):  # -> Optional[ModelSignature]
+        """
+        An optional definition of the expected inputs to and outputs from a model object, defined
+        with both field names and data types. Signatures support both column-based and tensor-based
+        inputs and outputs.
+
+        :getter: Retrieves the signature of a model instance iff the model was saved with a
+            signature definition.
+        :setter: Sets a signature to a model instance.
+        :type: Optional(ModelSignature)
```
Comment (harupy) [inline mlflow/models/model.py:369]:
```suggestion
        :type: Optional[ModelSignature]
```

File: mlflow/models/model.py [hunk 7]
```diff
@@ -298,16 +346,29 @@ def metadata(self) -> Optional[Dict[str, Any]]:
...
+        :return: Optional[ModelSignature]
```
Comment (harupy) [inline mlflow/models/model.py:370]: Do we need `:return:`?

**Review by harupy — APPROVED** [review]
LGTM once the remaining comments are addressed :)

**Pull request merged**